sdk/cosmosdb/cosmos/src/queryIterator.ts

```diff
@@ -161,19 +161,20 @@ export class QueryIterator<T> {
       try {
         response = await this.fetchAllInternal(diagnosticNode);
       } catch (error: any) {
-        if (this.clientContext.enableEncryption) {
+        if (this.container && this.clientContext.enableEncryption) {
           await this.container.ThrowIfRequestNeedsARetryPostPolicyRefresh(error);
         }
-        this.handleSplitError(error);
```

Comment:
why is this changed?

Reply:
added this in last PR by mistake. handleSplitError() is getting called in fetchAllInternal. https://github.com/Azure/azure-sdk-for-js/blob/85382568a328f8d04be2644b92c490aab561fbc0/sdk/cosmosdb/cosmos/src/queryIterator.ts#L139